x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/key.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiButtonIcon } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface Props {
+  copy: () => void;
+  toggleIsHidden: () => void;
+  isHidden: boolean;
+  text: string;
+}
+
+export const Key: React.FC<Props> = ({ copy, toggleIsHidden, isHidden, text }) => {
+  const icon = isHidden ? 'eye' : 'eyeClosed';
+  return (
+    <>
+      <EuiButtonIcon
+        onClick={copy}
+        iconType="copyClipboard"
+        aria-label={i18n.translate('xpack.enterpriseSearch.appSearch.credentials.copyApiKey', {
+          defaultMessage: 'Copy API Key to clipboard',
+        })}
+      />
+      <EuiButtonIcon
+        onClick={toggleIsHidden}
+        iconType={icon}
+        aria-label={i18n.translate(
+          'xpack.enterpriseSearch.appSearch.credentials.toggleApiEndpoint',
+          {
+            defaultMessage: 'Toggle API Key visibility',
+          }
+        )}
+      />
+      {text}
+    </>
```

Comment:
Accessibility/screen-reader suggestions:

```js
  const hideIcon = isHidden ? 'eye' : 'eyeClosed';
  const hideIconLabel = isHidden
    ? i18n.translate('xpack.enterpriseSearch.appSearch.credentials.showApiKey', {
        defaultMessage: 'Show API Key',
      })
    : i18n.translate('xpack.enterpriseSearch.appSearch.credentials.hideApiKey', {
        defaultMessage: 'Hide API Key',
      });

  return (
    <>
      // etc.
      <EuiButtonIcon
        onClick={toggleIsHidden}
        iconType={hideIcon}
        aria-label={hideIconLabel}
        aria-pressed={!isHidden}
      />
      {text}
    </>
  );
```

The conditional aria-label as well as aria-pressed helps inform screen-readers a tiny bit more as to what state the text is in without them having to navigate over to check.

I have a diff here for easy picking: https://github.com/constancecchen/kibana/commit/98e0f7e971f4a2aefd1a0e42bcefe0d33626ad96

Reply:
Good call on this one, thanks

Reply:
a42ba301512e99e569ad7e2fbb4869f80a54866b